File: ruby-neo4j-driver/test_social.rb
require "minitest/autorun"
require "./social"

class TestSocial < Minitest::Test
    def test_insert_friends
        sg = Social::Graph.new
        result = sg.insert_friends()
        assert_equal result.rows, []
    end

    def test_common_friends
        sg = Social::Graph.new
        result = sg.common_friends()
        assert_equal result, ["Bob"]
    end

    def test_friends_of_a_friend
        sg = Social::Graph.new
        result = sg.friends_of_a_friend()
        assert_equal result, ["Anna"]
    end

    def test_connecting_paths
        sg = Social::Graph.new
        result = sg.connecting_paths()
        assert_equal result, ["Joe", "Sally", "Anna", "Jim", "Billy"]
    end
end

File: python-neo4j-driver/test_social.py
import pytest
import social

def test_insert_friends():
    assert social.insert_friends() == None

def test_common_friends():
    assert social.common_friends() == "Bob"

def test_friends_of_a_friend():
    assert social.friends_of_a_friend() == "Anna"

def test_connecting_paths():
    result = social.connecting_paths()
    assert len(result) == 4
